compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6325,7 +6325,7 @@ public void visit(BLangTypeConversionExpr conversionExpr) {
         }
 
         BType targetType = conversionExpr.targetType;
-        validateIsNotCastToAnImportedAnonType(targetType == null ? conversionExpr.type : targetType);
+        // validateIsNotCastToAnImportedAnonType(targetType == null ? conversionExpr.type : targetType);
```

Comment:
Created https://github.com/ballerina-platform/ballerina-lang/issues/28511 to track adding this back.